pkg/controller/machinehealthcheck/machinehealthcheck_controller.go

```diff
@@ -352,3 +377,71 @@ func isMaster(machine mapiv1.Machine, client client.Client) bool {
 	}
 	return false
 }
+
+// isMachineDisruptionAllowed returns true if the provided MachineDisruptionBudget allows any disruption
+func isMachineDisruptionAllowed(mdb *healthcheckingv1alpha1.MachineDisruptionBudget, maxDisruptedMachinSize int) bool {
+	if mdb.Status.ObservedGeneration < mdb.Generation {
+		glog.Warningf("The machine disruption budget %s is still being processed by the server", mdb.Name)
+		return false
+	}
+	if mdb.Status.MachineDisruptionsAllowed < 0 {
+		glog.Warningf("The machine disruption budget %s MachineDisruptionsAllowed is negative", mdb.Name)
+		return false
+	}
+	if len(mdb.Status.DisruptedMachines) > maxDisruptedMachinSize {
+		glog.Warningf("The machine disruption budget %s DisruptedMachines map too big - too many deletions not confirmed by MDB controller", mdb.Name)
+		return false
+	}
+	if mdb.Status.MachineDisruptionsAllowed == 0 {
+		glog.Warningf("Cannot remediate machine as it would violate the machine's disruption budget %s", mdb.Name)
+		return false
+	}
+
+	return true
+}
+
+func (r *ReconcileMachineHealthCheck) decrementMachineDisruptionsAllowed(machineName string, mdb *healthcheckingv1alpha1.MachineDisruptionBudget) error {
```

Comment:
can you please move this to the MDB controller

Reply:
done